### ISSUE TITLE
Support Visual Studio 2022

### ISSUE THREAD

**Issue by xchesh** [therealjohn/UnityProjectTemplatesAndSnippets#4]
Hello!

How about Visual Studio 2022 suport?

Thank you!

**Comment by therealjohn** [comment]
Yes! I've started a PR to include all of this project into the Visual Studio Tools for Unity. 

That will include VS 2022 support and no longer require installing this separate extension.

**Comment by jourdanputtshack** [comment]
Hello! 

Is there any update on this? And how will we be able to add this to our projects? (Visual Studio Installer?)

**Comment by therealjohn** [comment]
Working on getting it into VS 2022 as part of the Game development with Unity workload in the VS Installer
![image](https://user-images.githubusercontent.com/1501196/157719648-1789ef57-4461-4cb3-8086-88e5c959e25e.png)

You likely already have this installed if you are using VS and Unity together. As soon as we have a release target I will update here.

**Comment by nader-naderi** [comment]
> Working on getting it into VS 2022 as part of the Game development with Unity workload in the VS Installer ![image](https://user-images.githubusercontent.com/1501196/157719648-1789ef57-4461-4cb3-8086-88e5c959e25e.png)
> 
> You likely already have this installed if you are using VS and Unity together. As soon as we have a release target I will update here.

hey john, big fan of your work. you made our life so easier and smoother I appreciate it. 
after more than half of a month,  still working on the VS2022 support? if you could bring it sooner you will save our neck quickly thanks man.

**Comment by nader-naderi** [comment]
> 

Thanks man. yeah, I'm using the tools but your works are not there yet.

**Comment by imniko** [comment]
Any update on this?

**Comment by malino24** [comment]
Any update on this?

**Comment by somedeveloper00** [comment]
I'm also interested to know what happened. VS 2022 currently just contains 2 Unity templates, one is monobehavhour and another is *supposedly* an Editor template, but it's actually a scriptable object with a menu item 🙂
![image](https://github.com/therealjohn/UnityProjectTemplatesAndSnippets/assets/79690923/4248aca1-2d1c-4187-bfa2-9ced01ad51ff)
```csharp
using UnityEditor;
using UnityEngine;

namespace Assets
{
    public class NewEditorScript1 : ScriptableObject
    {
        [MenuItem("Tools/MyTool/Do It in C#")]
        static void DoIt()
        {
            EditorUtility.DisplayDialog("MyTool", "Do It in C# !", "OK", "");
        }
    }
}
```

how this mistake made it to main release is beyond me

**Comment by dmfors** [comment]
I have no experience in creating extensions for Visual Studio. However, after trying and failing several times, I managed to make extension work with Visual Studio 2022. The templates and snippets function without any errors. Here is my fork:
https://github.com/DmFors/UnityProjectTemplatesAndSnippets
